wavesrv/pkg/sstore/memops.go

```diff
@@ -43,11 +44,94 @@ func isIndicatorGreater(i1 string, i2 string) bool {
 	return screenIndicatorLevels[i1] > screenIndicatorLevels[i2]
 }
 
+type OpenAICmdInfoChatStore struct {
+	MessageCount int                                `json:"messagecount"`
+	Messages     []*packet.OpenAICmdInfoChatMessage `json:"messages"`
+}
+
 type ScreenMemState struct {
-	NumRunningCommands int               `json:"numrunningcommands,omitempty"`
-	IndicatorType      string            `json:"indicatortype,omitempty"`
-	CmdInputText       utilfn.StrWithPos `json:"cmdinputtext,omitempty"`
-	CmdInputSeqNum     int               `json:"cmdinputseqnum,omitempty"`
+	NumRunningCommands int                     `json:"numrunningcommands,omitempty"`
+	IndicatorType      string                  `json:"indicatortype,omitempty"`
+	CmdInputText       utilfn.StrWithPos       `json:"cmdinputtext,omitempty"`
+	CmdInputSeqNum     int                     `json:"cmdinputseqnum,omitempty"`
+	AICmdInfoChat      *OpenAICmdInfoChatStore `json:"aicmdinfochat,omitempty"`
+}
+
+func ScreenMemInitCmdInfoChat(screenId string) {
+	greetingMessagePk := &packet.OpenAICmdInfoChatMessage{
+		MessageID:           0,
+		IsAssistantResponse: true,
+		AssistantResponse: &packet.OpenAICmdInfoPacketOutputType{
+			Message: packet.OpenAICmdInfoChatGreetingMessage,
+		},
+	}
+	ScreenMemStore[screenId].AICmdInfoChat = &OpenAICmdInfoChatStore{MessageCount: 1, Messages: []*packet.OpenAICmdInfoChatMessage{greetingMessagePk}}
+}
+
+func ScreenMemClearCmdInfoChat(screenId string) {
+	MemLock.Lock()
+	defer MemLock.Unlock()
+	if ScreenMemStore[screenId] == nil {
+		ScreenMemStore[screenId] = &ScreenMemState{}
+	}
+	ScreenMemInitCmdInfoChat(screenId)
+}
+
+func ScreenMemAddCmdInfoChatMessage(screenId string, msg *packet.OpenAICmdInfoChatMessage) {
+	MemLock.Lock()
+	defer MemLock.Unlock()
+	if ScreenMemStore[screenId] == nil {
+		ScreenMemStore[screenId] = &ScreenMemState{}
+	}
+	if ScreenMemStore[screenId].AICmdInfoChat == nil {
+		log.Printf("AICmdInfoChat is null, creating")
+		ScreenMemInitCmdInfoChat(screenId)
+	}
+
+	CmdInfoChat := ScreenMemStore[screenId].AICmdInfoChat
+	CmdInfoChat.Messages = append(CmdInfoChat.Messages, msg)
+	CmdInfoChat.MessageCount++
+}
+
+func ScreenMemGetCmdInfoMessageCount(screenId string) int {
+	MemLock.Lock()
+	defer MemLock.Unlock()
+	if ScreenMemStore[screenId] == nil {
+		ScreenMemStore[screenId] = &ScreenMemState{}
+	}
+	if ScreenMemStore[screenId].AICmdInfoChat == nil {
+		ScreenMemInitCmdInfoChat(screenId)
+	}
+	return ScreenMemStore[screenId].AICmdInfoChat.MessageCount
+}
+
+func ScreenMemGetCmdInfoChat(screenId string) *OpenAICmdInfoChatStore {
+	MemLock.Lock()
+	defer MemLock.Unlock()
+	if ScreenMemStore[screenId] == nil {
+		ScreenMemStore[screenId] = &ScreenMemState{}
+	}
+	if ScreenMemStore[screenId].AICmdInfoChat == nil {
+		ScreenMemInitCmdInfoChat(screenId)
+	}
+	return ScreenMemStore[screenId].AICmdInfoChat
```

Comment:
I think you're going to need to make a deep copy here to avoid any weird concurrency issues.

Reply:
addressed